enclave/crypto/CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+if (UNIX OR USE_CLANGW)
+    set(PLATFORM_SRC
+        ../../common/sgx/rand.S)
+elseif (WIN32)
+    set(PLATFORM_SRC
+        ../../common/sgx/rand.asm)
+endif()
+
+add_library(oecrypto STATIC
+    ../../common/asn1.c
+    ../../common/cert.c
+    ../../common/kdf.c
+    asn1.c
+    cert.c
+    crl.c
+    ec.c
+    entropy.c
+    cmac.c
+    hmac.c
+    key.c
+    random.c
+    rsa.c
+    sha.c
+    ${PLATFORM_SRC})
+
+maybe_build_using_clangw(oecrypto)
+
+target_include_directories(oecrypto PRIVATE
+    ${PROJECT_SOURCE_DIR}/include/openenclave/corelibc)
+
+target_link_libraries(oecrypto PUBLIC
+    oe_includes
+    mbedcrypto)
+
+if (OE_PREFIX_MBEDTLS_SYMBOLS)
```

Comment:
This logic is a little dangerous and is probably going to result in a "caching" issue. At some point someone is going to change the value of this option and not do a clean build and things won't work. It would be a major pain probably to support that scenario, so I'm not suggesting we should, just maybe leave a comment the optoin that if it's changed a clean build needs to happen. Otherwise we need to setup up some dependencies really carefully.

Reply:
Would it work if the library file name changes?
like this:
```
if (OE_PREFIX_MBEDTLS_SYMBOLS)
   set_target_properties(oecrypto PROPERTIES OUTPUT_NAME "oecrypto2")
   # objcopy, etc
endif()

install(TARGETS oecrypto EXPORT openenclave-targets
  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}/openenclave/enclave
  RENAME liboecrypto.a)
```

Reply:
Should the target be linking against oe_includes?

Reply:
@gupta-ak I don't think so, the main problem is conditionally adding the post-build step. If that script gets executed, then you turn OFF `OE_PREFIX_MBEDTLS_SYMBOLS`, you wouldn't _expect_ the script to have been executed, but I _don't think_ CMake is going to detect the removal of the post-build step and rebuild oecrypto without running the script. I think it'll just not update it, and then it'll have the symbols renamed despite it being OFF.

I'm not 100% certain this is going to be a problem, but I needed to point it out because sometimes CMake does weird things and it's usually around conditional logic like this.

Reply:
@radhikaj The target technically doesn't need to. `mbedcrypto` links against `oecore`, which exports `oe_includes` as a public, so it already has it. I can remove it, so the cmake graph is cleaner.

@andschwa Turns out that CMake is smart enough and it actually rebuilds!